x/incentives/keeper/genesis_test.go

```diff
@@ -75,41 +175,112 @@ func TestIncentivesInitGenesis(t *testing.T) {
 	validateGenesis := types.DefaultGenesis().Params.Validate()
 	require.NoError(t, validateGenesis)
 
-	// create coins, lp tokens with lockup durations, and a gauge for this lockup
-	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10000)}
 	startTime := time.Now()
-	distrTo := lockuptypes.QueryCondition{
-		LockQueryType: lockuptypes.ByDuration,
-		Denom:         "lptoken",
-		Duration:      time.Second,
-	}
-	gauge := types.Gauge{
-		Id:                1,
-		IsPerpetual:       false,
-		DistributeTo:      distrTo,
-		Coins:             coins,
-		NumEpochsPaidOver: 2,
-		FilledEpochs:      0,
-		DistributedCoins:  sdk.Coins(nil),
-		StartTime:         startTime.UTC(),
-	}
+	expectedGauges := expectedGauges(startTime)
+	// distrToNoLock denom gets added post creation when being called via createGauge, but
+	// we are manually creating the gauges here so we need to add it manually
+	expectedGauges[3].DistributeTo.Denom = "no-lock/e/1"
 
 	// initialize genesis with specified parameter, the gauge created earlier, and lockable durations
 	app.IncentivesKeeper.InitGenesis(ctx, types.GenesisState{
 		Params: types.Params{
 			DistrEpochIdentifier: "week",
 		},
-		Gauges: []types.Gauge{gauge},
+		Gauges: expectedGauges,
 		LockableDurations: []time.Duration{
 			time.Second,
 			time.Hour,
 			time.Hour * 3,
 			time.Hour * 7,
 		},
+		GroupGauges: expectedGroupGauges,
+		Groups:      expectedGroups,
 	})
 
 	// check that the gauge created earlier was initialized through initGenesis and still exists on chain
 	gauges := app.IncentivesKeeper.GetGauges(ctx)
-	require.Len(t, gauges, 1)
-	require.Equal(t, gauges[0], gauge)
+	require.Len(t, gauges, len(expectedGauges))
+
+	// duration gauge
+	require.Equal(t, expectedGauges[2], gauges[0])
+
+	// no lock gauge
+	require.Equal(t, expectedGauges[3], gauges[1])
+
+	// pool 1 gauge
+	require.Equal(t, expectedGauges[0], gauges[2])
+
+	// pool 2 gauge
+	require.Equal(t, expectedGauges[1], gauges[3])
+
+	// group gauge
+	groupGauges, err := app.IncentivesKeeper.GetAllGroupsGauges(ctx)
+	require.NoError(t, err)
+	require.Len(t, groupGauges, len(expectedGroupGauges))
+	require.Equal(t, expectedGroupGauges, groupGauges)
+
+	// check group
+	groups, err := app.IncentivesKeeper.GetAllGroups(ctx)
+	require.NoError(t, err)
+	require.Len(t, groups, 1)
+	require.Equal(t, expectedGroups, groups)
+}
+
+func createAllGaugeTypes(t *testing.T, app *osmoapp.OsmosisApp, ctx sdk.Context, addr sdk.AccAddress, coins sdk.Coins, startTime time.Time) {
+	// create a gauge that distributes coins to earlier created LP token and duration
+	_, err := app.IncentivesKeeper.CreateGauge(ctx, true, addr, coins, distrToByDuration, startTime, 1, 0)
+	require.NoError(t, err)
+
+	// create a gauge that distributes coins to earlier created LP token and duration
```

Comment:
isn't the one below a no-lock gauge?

Reply:
Yeah my mistake, was placeholder comments. Changed the comments here https://github.com/osmosis-labs/osmosis/pull/6565/commits/7829766d1f9af13b71745ad52b08fc2dcc2eb8c2